### PR TITLE
Add interface for specifying the offset of struct members in JitBuilder

### PR DESCRIPTION
**Pull request by Leonardo2718**
This interface provides a way of resolving issue #376 by allowing the user to specify offsets in such a way that they match those of the C/C++ compiler.

It also provides a workaround for issue #390, since the offset of the an inner struct can be simply specified. However, it does not address the issue of accessing the member. This will have to be resolved by another PR.

### CODE REVIEW

File: compiler/ilgen/TypeDictionary.cpp [hunk 1]
```diff
@@ -158,8 +158,10 @@ class StructType : public TR::IlType
       { }
 
    TR::DataType getPrimitiveType()                 { return TR::Address; }
+   void Close(unsigned int finalSize)                { TR_ASSERT(_size <= finalSize, "Final size %d of struct %s is less than its current size %d\n", finalSize, _name, _size); _size = finalSize; _closed = true; };
```
Comment (mstoodle) [inline compiler/ilgen/TypeDictionary.cpp]: Shouldn't this be a size_t ?

File: compiler/ilgen/TypeDictionary.cpp [hunk 2]
```diff
@@ -158,8 +158,10 @@ class StructType : public TR::IlType
...
    void Close()                                      { _closed = true; };
 
+   void AddField(const char *name, TR::IlType *fieldType, uint32_t offset);
```
Comment (mstoodle) [inline compiler/ilgen/TypeDictionary.cpp]: shouldn't the offset be size_t as well?

File: compiler/ilgen/TypeDictionary.cpp [hunk 3]
```diff
@@ -372,6 +391,17 @@ TypeDictionary::DefineStruct(const char *structName)
    }
 
 void
+TypeDictionary::DefineField(const char *structName, const char *fieldName, TR::IlType *type, uint32_t offset)
```
Comment (mstoodle) [inline compiler/ilgen/TypeDictionary.cpp]: offset should be size_t

File: compiler/ilgen/TypeDictionary.cpp [hunk 4]
```diff
@@ -177,6 +179,23 @@ class StructType : public TR::IlType
    };
 
 void
+StructType::AddField(const char *name, TR::IlType *typeInfo, uint32_t offset)
```
Comment (mstoodle) [inline compiler/ilgen/TypeDictionary.cpp]: size_t for offset

File: compiler/ilgen/TypeDictionary.cpp [hunk 5]
```diff
@@ -392,6 +422,15 @@ TypeDictionary::GetFieldType(const char *structName, const char *fieldName)
    }
 
 void
+TypeDictionary::CloseStruct(const char *structName, unsigned int finalSize)
```
Comment (mstoodle) [inline compiler/ilgen/TypeDictionary.cpp]: finalSize should be size_t

**Comment by Leonardo2718** [comment]
Done. I have also changed the type in a few other locations that were previously unchanged to keep the types consistent.

**Pull request merged**